extension/src/NABfunctions.ts

```diff
@@ -1121,3 +1123,124 @@ async function handleInvalidXmlError(error: unknown): Promise<void> {
     error.length
   );
 }
+
+export function getHoverText(
+  document: vscode.TextDocument,
+  position: vscode.Position
+): vscode.MarkdownString[] {
+  const settings = SettingsLoader.getSettings();
+  if (!settings.enableTranslationsOnHover) {
+    return [];
+  }
+
+  const returnValues = [];
+  const selectedLineNo = position.line;
+
+  const navObj = ALParser.getALObjectFromText(document.getText(), true);
+  if (!navObj) {
+    console.log(`Could not parse file ${document.fileName} as an al object`);
+    return [];
+  }
+  const mlObjects = navObj.getAllMultiLanguageObjects({
+    onlyForTranslation: true,
+  });
+  const selectedMlObject = mlObjects?.filter(
+    (x) => x.startLineIndex === selectedLineNo
+  );
```

Comment:
Use find? :)

Reply:
And check for undefined on line 1150

Reply:
Ah, maybe you need to assert that there's no multiples found? Not really used to those big brain parts of the extension ;)

Reply:
Yes, if we got more than 1 hit it's an unsupported case and we should just exit.